csrc/multidevice/communicator.cpp

```diff
@@ -102,16 +127,37 @@ c10::intrusive_ptr<c10d::Backend> createBackend(
   TORCH_CHECK(false, "no distributed backend available");
 }
 
-Communicator::Communicator(CommunicatorBackend backend, RankType server_rank)
-    : rank_(0), size_(0), local_rank_(0), local_size_(0) {
+Communicator::Communicator(
+    CommunicatorBackend backend,
+    RankType server_local_rank)
+    : is_available_(false),
+      rank_(0),
+      size_(0),
+      local_rank_(0),
+      local_size_(0),
+      master_port_(0) {
   // retrieves rank and communicator size
-  int status = parseEnv(rank_, size_, local_rank_, local_size_);
-  TORCH_CHECK(status == 0, "distributed configuration is not available");
+  is_available_ = parseEnv(
+      rank_, size_, local_rank_, local_size_, master_addr_, master_port_);
+
+  if (!is_available_) {
+    return;
+  }
 
   // creates the backend
   c10d::TCPStoreOptions store_opts;
-  store_opts.isServer = (rank_ == server_rank) ? true : false;
-  auto store = c10::make_intrusive<c10d::TCPStore>("localhost", store_opts);
+  {
+    char hostname[HOST_NAME_MAX]; // NOLINT (modernize-avoid-c-arrays)
+    gethostname(hostname, HOST_NAME_MAX);
+    struct hostent* master_host = gethostbyname(master_addr_.c_str());
```

Comment:
Does this need any header?

Reply:
yes, `<netdb.h>`, which is included here through the inclusion chain, but let me include it explicitly